src/main/java/picard/sam/SamErrorMetric/SamErrorReadFilterCriterion.java

```diff
@@ -0,0 +1,45 @@
+package picard.sam.SamErrorMetric;
+
+/**
+ * Class holding a criterion to be used in the SamErrorReadFilter class
+ * @param <T>
+ */
+public abstract class SamErrorReadFilterCriterion<T extends Comparable<T>> {
+    public final SamErrorReadFilter.Comparator comparator;
+    public final T value;
+    private boolean satisifed = false;
+
+    /**
+     * Creates a criterion object
+     * @param comparator Comparison operation to perform
+     * @param value Value to compare with to satisfy the criterion
+     */
+    public SamErrorReadFilterCriterion(final SamErrorReadFilter.Comparator comparator, final T value) {
+        this.comparator = comparator;
+        this.value = value;
+    }
+
+    /**
+     * Method to be called in order to check whether this criterion is satisfied
+     * @param stratus Object to check whether it satisfies the criterion
+     */
+    public abstract void checkCriterion(Object stratus);
```

Comment:
why does stratus not have a type? what are you going to put there?

Reply:
Since stratifiers can output different types, I wanted to keep it as generic as possible. I changed it now to be more strongly typed.